public/friendlycode/js/fc/publish.js

```diff
@@ -0,0 +1,172 @@
+define(function(require) {
+  var $ = require("jquery");
+  var bramble;
+  var csrfToken;
+  var publishBtn;
+  var updatePublishBtn;
+  var unpublishBtn;
+  var projectDesc;
+  var projectPublic;
+  var url;
+  var isPublic = true;
+
+  function setPublicGalleryToggle() {
+    if(isPublic) {
+      $({ cx: projectPublic.attr("cx") })
+      .animate({ cx: 31 }, {
+        duration: 250,
+        step: function(now) {
+          projectPublic.attr("cx", now);
+        }
+      });
+      projectPublic.css("fill", "#06a050");
+    }
+    else {
+      $({ cx:projectPublic.attr("cx") })
+      .animate({ cx: 11 }, {
+        duration: 250,
+        step: function(now) {
+          projectPublic.attr("cx", now);
+        }
+      });
+      projectPublic.css("fill", "#7C7C7C");
+    }
+  }
+
+  function unpublish() {
+    unpublishBtn.off("click", unpublish);
+    unpublishBtn.children("span").text("Unpublishing...");
+
+    var project = {
+      description: projectDesc.val() || " ",
+      public: isPublic,
+      dateUpdated: (new Date()).toISOString()
+    };
+
+    var request = $.ajax({
+      contentType: "application/json",
+      headers: {
+        "X-Csrf-Token": csrfToken,
+        "Accept": "application/json"
+      },
+      type: "PUT",
+      url: url + "/unpublish",
+      data: JSON.stringify(project)
+    });
+    request.done(function() {
+      if(request.status !== 200) {
+        console.error("[Bramble] Server was unable to unpublish project, responded with status ", request.status);
+        unpublishBtn.off("click", unpublish).on("click", unpublish);
```

Comment:
yep good catch! I'm going to probably disable the publish button as well so that we don't run into timing issues if you click both buttons at the same time